src/components/pages/PeoplePage.tsx

```diff
@@ -0,0 +1,52 @@
+import { Loader } from '../Loader';
+import { PeopleTable } from '../Loader/PeopleTable';
+import { getPeople } from '../services/api';
+import { useEffect, useState } from 'react';
+import { Person } from '../../types';
+
+export const PeoplePage = () => {
+  const [people, setPeople] = useState<Person[]>([]);
+  const [isLoading, setIsLoading] = useState(true);
+  const [isError, setIsError] = useState(false);
+
+  useEffect(() => {
+    getPeople()
+      .then(data => {
+        const peopleByName: { [key: string]: Person } = {};
+
+        data.forEach(person => (peopleByName[person.name] = person));
+        const dataUpdated = data.map(p => {
+          return {
+            ...p,
+            mother: peopleByName[p.motherName || ''],
+            father: peopleByName[p.fatherName || ''],
+          };
```

Comment:
Code quality: This block of code is a bit hard to read and understand. It would be better to separate the logic of mapping the people's data into a separate function. This will make your code cleaner and more maintainable.